src/corehost/cli/hostpolicy/hostpolicy.cpp

```diff
@@ -392,8 +423,58 @@ int corehost_libhost_init(hostpolicy_init_t &hostpolicy_init, const pal::string_
 
 namespace
 {
-    int get_coreclr_delegate(const std::shared_ptr<coreclr_t> &coreclr, coreclr_delegate_type type, void** delegate)
+    int load_runtime_for_context(const context_handle instance)
```

Comment:
I think I'm missing something:
* hostfxr is supposed to call `load_runtime_for_context` before calling any of the run APIs (at least it seems that's what it's doing) - we probably need that to handle the synchronization correctly around the "run app" (Since once called it blocks till the end).
* If that is the case, then why are we calling `get_or_create_coreclr` from the `run_app_for_context` below (the the get delegate version as well)?

Reply:
Yes, that is the case. They should always be using an already loaded runtime... I missed updating `run_app/get_delegate_for_context` after splitting out `load_runtime_for_context`.